bugbug/bugzilla.py

```diff
@@ -93,50 +94,78 @@ def bughandler(bug):
-            new_bugs[bug_id] = dict()
+def get(bug_ids):
+    """Function to retrieve Bug Information including history, attachment, comments using Bugzilla REST API.
 
-        new_bugs[bug_id]["comments"] = bug["comments"]
-
-    def attachmenthandler(bug, bug_id):
-        bug_id = int(bug_id)
-
-        if bug_id not in new_bugs:
-            new_bugs[bug_id] = dict()
-
-        new_bugs[bug_id]["attachments"] = bug
-
-    def historyhandler(bug):
-        bug_id = int(bug["id"])
+    :param bug_ids: find bug information for these `bug_ids`
+    :type bug_ids: list of integers or str or int
+    :return: dict with key as `id`(int) of a bug,  and values as bug_information
+    :rtype: dict
+    """
 
-        if bug_id not in new_bugs:
-            new_bugs[bug_id] = dict()
+    def filter_keys(array_of_dict_values, required_fields):
+        for a_value in array_of_dict_values:
+            to_be_deleted_fields = set(a_value.keys()).difference(set(required_fields))
+            for extra_field in to_be_deleted_fields:
+                del a_value[extra_field]
+        return array_of_dict_values
 
-        new_bugs[bug_id]["history"] = bug["history"]
+    new_bugs = {}
 
-    Bugzilla(
-        ids_or_query,
-        bughandler=bughandler,
-        commenthandler=commenthandler,
-        comment_include_fields=COMMENT_INCLUDE_FIELDS,
-        attachmenthandler=attachmenthandler,
-        attachment_include_fields=ATTACHMENT_INCLUDE_FIELDS,
-        historyhandler=historyhandler,
-    ).get_data().wait()
+    if isinstance(bug_ids, list):
+        # Expected Format
+        bug_ids_list = list(set(map(int, bug_ids)))
+
+    elif isinstance(bug_ids, six.string_types):
+        bug_ids_list = [int(bug_ids)]
+
+    elif isinstance(bug_ids, int):
+        bug_ids_list = [bug_ids]
+    else:
+        raise ValueError(
+            "Invalid datatype for the parameter-bug_ids with value- {bug_ids} and datatype-{type(bug_ids)}"
+        )
+
+    new_bugs = dict()
+    for a_bug_id in bug_ids_list:
+        params_for_custom_fields = {
+            "id": a_bug_id,
+            "include_fields": "_default,history,comments,attachments",
+        }
+        response = utils.get_session("bugzilla").get(
+            "https://bugzilla.mozilla.org/rest/bug", params=params_for_custom_fields
+        )
+        response.raise_for_status()
+
+        a_bug_info = response.json()
+        a_bug_info = a_bug_info["bugs"]
+
+        assert len(a_bug_info) == 1
+
+        a_bug_info = a_bug_info[0]
+
+        # Realignment of results to cope with previous versions of data manipulation
+        bug_id_of_a_bug_info = int(a_bug_info["id"])
+
+        # Delete other fields from comments
+        current_comments_array = a_bug_info["comments"]
+        current_comments_array = filter_keys(
+            array_of_dict_values=current_comments_array,
+            required_fields=COMMENT_INCLUDE_FIELDS,
+        )
+        a_bug_info["comments"] = current_comments_array
```

Comment:
Can't we pass the include fields to Bugzilla instead of manually removing the fields after the fact?

Reply:
include_fields is assigned `"_default,history,comments,attachments"`. We don`t have fine grained control on sub-fields of `comments`, 'attachments`.  So, the manual deletion is implemented here.

Reply:
Fixed in this commit : https://github.com/mozilla/bugbug/pull/1575/commits/15b460fe9eb94af905461d114a45860c47c92be8

Reply:
Oh, if that's the case, then unfortunately we cannot do this, as we'd spend too much time and resources downloading the attachments' contents.

Reply:
1) Using "include_fields": "_default,history,comments,attachments", We are getting HISTORY, COMMENTS and ATTACHMENTS of a bug source: https://bmo.readthedocs.io/en/latest/api/core/v1/bug.html#get-bug
2) The HISTORY is an array, COMMENTS is an array, ATTACHMENTS is an array. 
3) ATTACHMENTS will have 15 fields, But we need only 7 fields. Even if we dont delete them, there will be no impact in running the successive codes.
4) The fields deletion is done to sync with previous version of code.
5) Similarly, COMMENTS will have 11 fields, but we need 5 fields.
6) Alternatively, For a BATCH of Bugs, 
    - We can get BUGS  _default fields, history fields,comments fields using REST API.
    - Using separate call for each bug, we can get Custom Attachment Fields which will save from getting attachment data.
    - Example https://bugzilla.mozilla.org/rest/bug/1633278/attachment?include_fields=id,is_obsolete,flags,is_patch,creator,content_type,creation_time 
    - Here we have to make request for each bug.
    - Getting attachment meta-information  can be retrieved per bug, instead of batches of bugs. 

7) Please give suggestions on how to proceed further.